rpm/bundle-rust-npm.py

```diff
@@ -0,0 +1,251 @@
+        contents = file.readlines()
+
+    log.info(f"Remove '{SPECFILE_COMMENT_LINE}' content from {spec_file}")
+    for line in contents:
+        if line == START_LINE:
+            remove_lines = True
+            log.debug(f"Remove '{START_LINE}' from {spec_file}")
+        elif line == END_LINE:
+            remove_lines = False
+            cleaned = True
+            log.debug(f"Remove '{END_LINE}' from {spec_file}")
+        elif not remove_lines:
+            result.append(line)
+        else:
+            log.debug(f"Remove '{line}' from {spec_file}")
+
+    with open(spec_file, "w") as file:
+        file.writelines(result)
+    return cleaned
+
+
+def write_provides_bundled(provides_lines: List[str], spec_file: str, cleaned: bool):
+    """Writes bundled package information to the spec file. 
+    Includes generated 'Provides' lines and marks the section for easy future modification.
+    """
+    # Find a line index where 'Provides' ends
+    with open(spec_file, "r") as file:
+        spec_file_lines = file.readlines()
+    last_provides = -1
+    for i in range(0, len(spec_file_lines)):
+        if spec_file_lines[i].startswith("%description"):
+            break
+        if spec_file_lines[i].startswith("Provides:"):
+            last_provides = i
+
+    # Insert the generated 'Provides' to the specfile
+    log.info(f"Add the fresh '{SPECFILE_COMMENT_LINE}' content to {spec_file}")
+    i = last_provides + 2
+    spec_file_lines.insert(i, START_LINE)
+    for line in sorted(provides_lines):
+        i = i + 1
+        log.debug(f"Adding '{line[:-1]}' as a line {i} to buffer")
+        spec_file_lines.insert(i, line)
+    i = i + 1
+    spec_file_lines.insert(i, END_LINE)
+
+    # Insert an empty line if we haven't cleaned the old content
+    # (as the old content already has an extra empty line that wasn't removed)
+    if not cleaned:
+        i = i + 1
+        spec_file_lines.insert(i, "\n")
+
+    log.debug(f"Commit the buffer to {spec_file}")
+    with open(spec_file, "w") as file:
+        file.writelines(spec_file_lines)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+    log = setup_script_logger('bundle-rust-npm', args.verbose)
+
+    log.debug("389-ds-base Rust Crates and Node Modules to Bundled Downstream Specfile tool")
+    log.debug(f"Called with: {args}")
+
+    if args.backup_specfile:
+        backup_specfile(args.spec_file)
+
+    if not os.path.isdir(args.cargo_path):
+        log.error(f"Path {args.cargo_path} does not exist or is not a directory")
+        sys.exit(1)
+    if not os.path.isdir(args.npm_path):
+        log.error(f"Path {args.npm_path} does not exist or is not a directory")
+        sys.exit(1)
+
+    cleaned = clean_specfile(args.spec_file)
+
+    rust_output = run_cmd(["cargo", "license", "--json", "--current-dir", args.cargo_path])
+    npm_output = run_cmd(["license-checker", "--json", "--start", args.npm_path])
+
+    if rust_output is None or npm_output is None:
```

Comment:
For me it fails before it reaches this code:
```
DS_SPECFILE=$HOME/src/fedora/389-ds-base/389-ds-base.spec make -f rpm.mk bundle-rust-npm
python3 rpm/bundle-rust-npm.py src/ src/cockpit/389-console/ /home/vashirov/src/fedora/389-ds-base/389-ds-base.spec --backup-specfile
Backing up file /home/vashirov/src/fedora/389-ds-base/389-ds-base.spec to /home/vashirov/src/fedora/389-ds-base/389-ds-base.spec.20240124_221456
Remove 'Bundled cargo crates list' content from /home/vashirov/src/fedora/389-ds-base/389-ds-base.spec
Traceback (most recent call last):
  File "/home/vashirov/src/ds/rpm/bundle-rust-npm.py", line 233, in <module>
    npm_output = run_cmd(["license-checker", "--json", "--start", args.npm_path])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vashirov/src/ds/rpm/bundle-rust-npm.py", line 75, in run_cmd
    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib64/python3.12/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'license-checker'
make: *** [rpm.mk:48: bundle-rust-npm] Error 1
```

Perhaps we should check if `cargo-license` and `license-checker` are in $PATH before running them?